mjcf_to_sdformat/mjcf_to_sdformat/converters/link.py

```diff
@@ -0,0 +1,38 @@
+# Copyright (C) 2022 Open Source Robotics Foundation
+#
+# Licensed under the Apache License, version 2.0 (the "License")
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from sdformat_mjcf.converters.geometry import (add_mjcf_visual_to_sdf,
+                                               add_mjcf_collision_to_sdf)
+
+import sdformat as sdf
+
+NUMBER_OF_SDF_LINK = 0
+
+
+def add_mjcf_link_to_sdf(geom):
```

Comment:
Should this take a MJCF body instead of a geom?

Reply:
I don't see any changes. Is this resolved?

Reply:
https://github.com/gazebosim/gz-mujoco/pull/26/commits/831582a9e8485930cbea550241eded450ce4f016

Reply:
Ah, but I think `add_mjcf_body_to_sdf` should take an MJCF `body` as an argument, not a `geom`, and return an `sdformat.Link`.

Reply:
I made the logic the use `geom` instead of `body`. I changed the name of the method `add_mjcf_geom_to_sdf` https://github.com/gazebosim/gz-mujoco/pull/26/commits/3a0184d7c42ac2782a4e58d39c262b5b8fed9d26

Reply:
I still fail to understand why we are associating `geom` to links. There should be a 1 to 1 correspondence between SDFormat links and MJCF bodies.
Consider the following MJCF file:
```xml
<mujoco model="test">
  <compiler eulerseq="XYZ"/>
  <worldbody>
    <body name="base">
      <geom name="geom1" type="cylinder" size="0.8 0.01" />
      <geom name="geom2" type="box"  size="0.1 0.1 1.1" pos="0 0 0.1"/>
    </body>
  </worldbody>
</mujoco>
```
I would expect the resulting SDFormat to have 1 model with 1 link and that link to contain 2 collisions:

```xml
<world name="default">

  <!-- Some world parameters, such as gravity -->

  <model name="test">
    <link name="base">
      <collision name="collision_geom1" type="cylinder">
        <radius>0.8</radius>
        <length>0.02</length>
      </collision>

      <visual name="visual_geom1" type="cylinder">
        <radius>0.8</radius>
        <length>0.02</length>
      </visual>

      <collision name="collision_geom2" type="box">
        <pose>0 0 0.1 0 0 0</pose>
        <size>0.2 0.2 0.2</size>
      </collision>

      <visual name="visual_geom2" type="box">
        <pose>0 0 0.1 0 0 0</pose>
        <size>0.2 0.2 0.2</size>
      </visual>
    </link>
  </model>
</world>
```

Note: I added `visual_` or `collision_` as prefixes in the names of visuals and collisions as an example. We can come up with a different naming scheme if desired.

But what I get from running the `mjcf2sdformat` script has 2 models and each model has 1 link and each link has 1 collision and 1 visual.
```xml
<sdf version='1.9'>
  <world name='test'>
    <gravity>0 0 -9.8066499999999994</gravity>
    <magnetic_field>5.5644999999999998e-06 2.2875799999999999e-05 -4.2388400000000002e-05</magnetic_field>
    <wind>
      <linear_velocity>0 0 0</linear_velocity>
    </wind>
    <physics default='false' type='ode'>
      <max_step_size>0.001</max_step_size>
      <real_time_factor>1</real_time_factor>
      <max_contacts>20</max_contacts>
    </physics>
    <model name='model_0'>
      <static>false</static>
      <self_collide>false</self_collide>
      <allow_auto_disable>true</allow_auto_disable>
      <enable_wind>false</enable_wind>
      <pose>0 0 0 0 0 0</pose>
      <link name='cylinder_0'>
        <pose>0 0 0 0 0 0</pose>
        <inertial>
          <pose>0 0 0 0 0 0</pose>
          <mass>1</mass>
          <inertia>
            <ixx>1</ixx>
            <ixy>0</ixy>
            <ixz>0</ixz>
            <iyy>1</iyy>
            <iyz>0</iyz>
            <izz>1</izz>
          </inertia>
        </inertial>
        <enable_wind>false</enable_wind>
        <collision name='collision_0'>
          <pose>0 0 0 0 0 0</pose>
          <geometry>
            <cylinder>
              <radius>0.80000000000000004</radius>
              <length>0.02</length>
            </cylinder>
          </geometry>
          <surface>
            <contact>
              <collide_bitmask>255</collide_bitmask>
            </contact>
          </surface>
        </collision>
        <visual name='visual_0'>
          <pose>0 0 0 0 0 0</pose>
          <geometry>
            <cylinder>
              <radius>0.80000000000000004</radius>
              <length>0.02</length>
            </cylinder>
          </geometry>
          <cast_shadows>true</cast_shadows>
          <laser_retro>0</laser_retro>
          <transparency>0</transparency>
          <visibility_flags>4294967295</visibility_flags>
        </visual>
      </link>
    </model>
    <model name='model_1'>
      <static>false</static>
      <self_collide>false</self_collide>
      <allow_auto_disable>true</allow_auto_disable>
      <enable_wind>false</enable_wind>
      <pose>0 0 0 0 0 0</pose>
      <link name='box_1'>
        <pose>0 0 0 0 0 0</pose>
        <inertial>
          <pose>0 0 0 0 0 0</pose>
          <mass>1</mass>
          <inertia>
            <ixx>1</ixx>
            <ixy>0</ixy>
            <ixz>0</ixz>
            <iyy>1</iyy>
            <iyz>0</iyz>
            <izz>1</izz>
          </inertia>
        </inertial>
        <enable_wind>false</enable_wind>
        <collision name='collision_1'>
          <pose>0 0 0 0 0 0</pose>
          <geometry>
            <box>
              <size>0.10000000000000001 0.10000000000000001 1.1000000000000001</size>
            </box>
          </geometry>
          <surface>
            <contact>
              <collide_bitmask>255</collide_bitmask>
            </contact>
          </surface>
        </collision>
        <visual name='visual_1'>
          <pose>0 0 0 0 0 0</pose>
          <geometry>
            <box>
              <size>0.10000000000000001 0.10000000000000001 1.1000000000000001</size>
            </box>
          </geometry>
          <cast_shadows>true</cast_shadows>
          <laser_retro>0</laser_retro>
          <transparency>0</transparency>
          <visibility_flags>4294967295</visibility_flags>
        </visual>
      </link>
    </model>
  </world>
</sdf>


```